### ISSUE TITLE
Problem with unknown options

### ISSUE THREAD

**Issue by MartinKolarik** [babel/gobble-babel#2]
Since v3.0.0, 6to5 throws when you supply an unknown option, which means this module fails due to [this line](https://github.com/6to5/gobble-6to5/blob/9f5d08dbaffee57a51a3778cab925fe12e91bd65/index.js#L11). If there's no other way to configure gobble, the key should probably be deleted before passing options to 6to5.

**Comment by Rich-Harris** [comment]
Ah, shoot. Deleting the key is probably the way to go - the options object is cloned before the function is called (so that we can manipulate it in exactly this sort of way without consequence).

Though since the `accept` option is never used by plugins themselves but gobble's built-in map transformer, it might make sense to delete `accept` and `ext` from the default options object at startup as soon as the values have been read. Then plugins don't have to do anything.

The alternative (and arguably more 'correct') approach would be to structure the options like...

``` js
gobble( 'src' ).transform( '6to5', {
  to5Options: {
    blacklist: [ 'modules' ]
  }
});
```

...but I'm not really into that, it seems like overkill.

**Comment by MartinKolarik** [comment]
> it might make sense to delete accept and ext from the default options object at startup as soon as the values have been read. Then plugins don't have to do anything.

That makes sense. It's probably the best solution.

**Comment by sebmck** [comment]
Damn, I tried to catch all of these when I was updating all the plugins to 6to5 3.0.0. I'll add `accept` as a valid option as I've done with some of the other plugins that used custom options.

**Comment by sjmeverett** [comment]
@sebmck what about `ext`?

**Comment by sebmck** [comment]
@stewartml What about it?

**Comment by sjmeverett** [comment]
Sorry for the brevity.  Using `ext` I can change the extension from say `.es6` to `.js`; however babel complains about it.  I can fix this by either adding it as a valid option to babbel, or by adding a line to delete the option in gobble-babel.

I wonder though: surely it's the job of the plugins not to send bad options rather than babel to special case any options that might come in from plugins, or have I got the wrong end of the stick?

**Comment by sebmck** [comment]
@stewartml Yeah, `gobble-babel` should be removing all invalid options before passing it on to `babel`.

**Comment by sebmck** [comment]
The only reason I whitelisted some in the first place was to avoid having to go and update all the plugins as I'm not familiar at all with a large majority of the libraries that there are plugins for.

**Comment by sjmeverett** [comment]
I see, that's fair enough.  I could submit a patch for `gobble-babel` to delete `accept` and `ext`, or perhaps @Rich-Harris would know better all the keys that should be deleted before passing to `babel`.

**Comment by Rich-Harris** [comment]
Thanks guys, you're right about it not being babel's responsibility - I've chosen to fix this in gobble itself by treating `accept` and `ext` options with map transforms as 'special' (which they are) - https://github.com/gobblejs/gobble/pull/36. That way other plugins won't encounter this problem.

I agree, it makes sense to allow `.es6`, though personally I just use `.js` since a) it's just JavaScript and b) it really should be `.es2015` now :)

Have published gobble 0.7.5 and gobble-babel 4.0.1 with this change.

**Comment by sjmeverett** [comment]
Thanks for this.  You're right, I would just use `.js` myself too, but it's in a project where I'll avoid some potential issues if I make it obvious that there's some extra translation happening.
